controllers/admin/AdminImportController.php

```diff
@@ -3725,23 +4668,38 @@ public function storeContactImportOne($info, $shop_is_feature_active, $regenerat
-                $country->iso_code = Tools::strtoupper(Tools::substr($store->country, 0, 2)); // Default iso for country to create
+                // Default zone for country to create
+                $country->id_zone = 0;
+                // Default iso for country to create :
+                $country->iso_code = Tools::strtoupper(Tools::substr($store->country, 0, 2));
                 $country->contains_states = 0; // Default value for country to create
-                $lang_field_error = $country->validateFieldsLang(UNFRIENDLY_ERROR, true);
-                if (($field_error = $country->validateFields(UNFRIENDLY_ERROR, true)) === true &&
-                    ($lang_field_error = $country->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true &&
-                    !$validateOnly && // Do not move this condition: previous tests should be played always, but next ->add() test should not be played in validateOnly mode
-                    $country->add()) {
+                $country->need_identification_number = 0; // Default value for country to create
+                $langFieldsValidationResult = $country->validateFieldsLang(UNFRIENDLY_ERROR, true);
+                // FIXME needing such a comment about "&& !$validateOnly" position is a code smell. Refacto needed.
+                if (($fieldsValidationResult = $country->validateFields(UNFRIENDLY_ERROR, true)) === true
+                    && ($langFieldsValidationResult = $country->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true
+                    && !$validateOnly // Do not move this condition: previous tests should be played always, but next ->add() test should not be played in validateOnly mode
+                    && $country->add()
+                ) {
                     $store->id_country = (int)$country->id;
                 } else {
                     if (!$validateOnly) {
                         $default_language_id = (int)Configuration::get('PS_LANG_DEFAULT');
-                        $this->errors[] = sprintf($this->trans('%s cannot be saved', array(), 'Admin.Advparameters.Notification'), $country->name[$default_language_id]);
+                        $this->errors[] = $this->trans(
+                            '%s cannot be saved',
+                            array($country->name[$default_language_id]),
+                            'Admin.Advparameters.Notification'
+                        );
                     }
-                    if ($field_error !== true || isset($lang_field_error) && $lang_field_error !== true) {
-                        $this->errors[] = ($field_error !== true ? $field_error : '').(isset($lang_field_error) && $lang_field_error !== true ? $lang_field_error : '').
-                            Db::getInstance()->getMsgError();
+                    if ($fieldsValidationResult !== true
+                        || isset($langFieldsValidationResult) && $langFieldsValidationResult !== true
```

Comment:
- && is prevalent on ||
- I created a new line for this group
- but ok, I'll change this in next commit.